### ISSUE TITLE
Deleted models leave behind their Collection keys.

### ISSUE THREAD

**Issue by zTristan** [agrison/johm#5]
I'm using Redis Desktop Manager to look at the stored keys and noticed that JOhm was leaving behind a bunch of orphaned keys after the parent objects were being deleted. They all seemed to be Collection indexes.

Test case:
```
@Model
public class Test {
	@Id
	public Long id;
	@Attribute
	public String name;
	@CollectionList(of = TestMessage.class)
	public List<TestMessage> messages;
}

@Model
public class TestMessage {
	@Id
	public Long id;
	@Attribute
	public String message;
}

Test test = new Test();
test.name = "Test";
JOhm.save(test);

TestMessage testMessage = new TestMessage();
testMessage.message = "HelloWorld";
JOhm.save(testMessage);
test.messages.add(testMessage);

JOhm.delete(Test.class, test.id, true, true);

String key = String.format("Test:%d:messages", test.id);
boolean keyexists = _jedisPool.getResource().exists(key);
LOGGER.info(String.format("Key Exists: %s", keyexists?"YES":"NO"));
```

**Comment by dpoldrugo** [comment]
I have done some improvments in my fork, I think I already solved this issue. Give me few days to make a pull request because I have done other changes. Some of them are:

- @SupportAll - id removal from collection when deleting
- Sentinel support - so JOhm can be used with HA Redis/Sentinel

**Comment by agrison** [comment]
Sorry my Github mails were in my spam I did not notice your issue before!
I will look into it.

**Comment by agrison** [comment]
Only Arrays where supported while deleting a model and its children. I'm adding support for the other various collections (`CollectionList`, `CollectionSet` and so on).

Now @zTristan in your example, you're just checking that the key `Test:1:messages` has been deleted but not that `TestMessage:1` (which was contained in `Test:1:messages`) has been deleted also, is that the behaviour you were expecting at first sight ?

**Comment by agrison** [comment]
I just released version 0.6.7 including the latest pull requests from @dpoldrugo and the fix for issue #9 and #5.

**Comment by zTristan** [comment]
> Now @zTristan in your example, you're just checking that the key Test:1:messages has been deleted but not that TestMessage:1 (which was contained in Test:1:messages) has been deleted also, is that the behaviour you were expecting at first sight ?
@agrison yes, it was just the indexes for the collection. The object themselves were being cleaned up.

Thanks for the update! I'll pull down 0.6.7 and verify!
